packages/app/src/runner/event-manager.ts

```diff
@@ -611,15 +607,19 @@ export class EventManager {
         })
       }
 
-      await Cypress.backend('protocol:test:before:run:async', { id: test.id, title: test.title, wallClockStartedAt: test.wallClockStartedAt.getTime() })
+      await Cypress.backend('protocol:test:before:run:async', { id: attr.id, attempt: attr.currentRetry + 1, timestamp: attr.wallClockStartedAt.getTime() })
 
       Cypress.primaryOriginCommunicator.toAllSpecBridges('test:before:run:async', ...args)
     })
 
     Cypress.on('test:after:run', (test) => {
+      this.reporterBus.emit('test:after:run', test, Cypress.config('isInteractive'))
+
       if (this.studioStore.isOpen && test.state !== 'passed') {
         this.studioStore.testFailed()
       }
+
+      Cypress.backend('protocol:test:after:run', { id: test.id, attempt: test.currentRetry + 1, wallClockDuration: test.wallClockDuration, timestamp: test.wallClockStartedAt.getTime() + test.wallClockDuration })
```

Comment:
Have you looked at passing the timings down instead of calculating the timestamp? it would be `test.timings`

also since you are updating this file, this should really be updated to:
`Cypress.on('test:after:run', (attributes, test) => {` where attributes are test attributes and test is the Mocha Test Runnanble object.

Reply:
> Have you looked at passing the timings down instead of calculating the timestamp?

Yes, I looked at the `timings` and unfortunately it doesn't include the end time.

> Also just curious, any reason to hook in from the app events vs sending directly from Cypress when we emit [test:after:run](https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress.ts#L549)?

The `protocol` exists on the server not in the app, so this seemed like the most appropriate place to add it. I could call the backend from the runner but I'm not sure what benefit that would be.

> also since you are updating this file, this should really be updated to the following:

Yes, I can update the parameter name.

Reply:
> i'm not sure what benefit that would be.

Reply:
Renamed parameters: [b4b0dea](https://github.com/cypress-io/cypress/pull/26216/commits/b4b0dea1bed69ee9f8931389b1968b0c9312c77b)